### DISCUSSION TITLE
Go to definition PVL VS Code IDE

### DISCUSSION THREAD

**Discussion by pandasansgains** [utwente-fmt/vercors#1169] [Q&A] [answered]
Hello, @pieter-bos I start this discussion concerning the go to definition feature we discussed earlier this week.

We had discussed setting up a stage with resolution and parsing steps for PVL only .

My question is the following. How would i bind this stage to a certain command within this project which would take a file path argument and return the results of this stage only. (Could you point us towards examples within the vercors project ?)

(If this is not the way and there is a way to use the already existing commands but run this stage using an additional option that also would be fine)

Additionnally it will be necessary for us to compile/make a release of vercors so we can use it within the IDE when we will have modified (Or added this stage for example), so we can use it within the extension by setting the right bin path.  Could you also tell us a little more on how we could achieve the following ?

**Comment by pieter-bos** [comment, marked as answer]
You can add a mode to the options [here](https://github.com/utwente-fmt/vercors/blob/2a94d1ddf4fc804be6da61ce79fbbac76a1538bb/src/main/vct/options/types/Mode.scala) and [here](https://github.com/utwente-fmt/vercors/blob/2a94d1ddf4fc804be6da61ce79fbbac76a1538bb/src/main/vct/options/Options.scala#L81), the main method splits by mode [here](https://github.com/utwente-fmt/vercors/blob/2a94d1ddf4fc804be6da61ce79fbbac76a1538bb/src/main/vct/main/Main.scala#L96), typically a mode chains stages together to run them. There should be plenty of examples in the options for file arguments.

Instructions for compiling VerCors are [here](https://github.com/utwente-fmt/vercors?tab=readme-ov-file#building-from-source-code), or you can see about importing it into an IDE [here](https://github.com/utwente-fmt/vercors/wiki/IDE-Import) (but this can be a bit fiddly)
